Lib/zipfile.py

```diff
@@ -836,9 +837,11 @@ def __init__(self, fileobj, mode, zipinfo, pwd=None,
 
         if hasattr(zipinfo, 'CRC'):
             self._expected_crc = zipinfo.CRC
+            self._orig_crc = zipinfo.CRC
             self._running_crc = crc32(b'')
         else:
             self._expected_crc = None
+            self._orig_crc = None
```

Comment:
I would just add `self._orig_crc = self._expected_crc` after line 852. It is not needed if the stream is non-seekable.